services/galley/src/Galley/Data/Queries.hs

```diff
@@ -307,6 +307,9 @@ insertUserRemoteConv = "insert into user_remote_conv (user, conv_remote_domain,
 selectUserRemoteConvs :: PrepQuery R (Identity UserId) (Domain, ConvId)
 selectUserRemoteConvs = "select conv_remote_domain, conv_remote_id from user_remote_conv where user = ? order by conv_remote_domain"
 
+selectRemoteConvMembership :: PrepQuery R ([UserId], Domain, ConvId) (Identity UserId)
+selectRemoteConvMembership = "select user from user_remote_conv where user in ? and conv_remote_domain = ? and conv_remote_id = ?"
```

Comment:
IIRC, writing `in` queries in Cassandra just makes it do more network communication because now the cassandra node has to communicate with other nodes to gather data. It is better to do either a mapConcurrently or even sequential queries. But I could be wrong, @jschaul can you please help?

Reply:
Yes, see e.g. https://lostechies.com/ryansvihla/2014/09/22/cassandra-query-patterns-not-using-the-in-query-for-multiple-partitions/ and change this query to take a single UserId. `pooledMapConcurrentlyN 8` inside a Monad that allows that can be your friend.

(I feel like we should have a docs/developer/cassandra.md section for these things, they come up time and again)

Reply:
Thanks, I'll have a read and change the query accordingly.